composite_broker/src/broker/tree.rs

```diff
@@ -0,0 +1,251 @@
+pub mod tree {
```

Comment:
I think you went away from this implementation, right? I'll ignore it for now. If I'm wrong, and you're using it, I'll go back and have a look!